### PR TITLE
54622 qualification filters

### PR DESCRIPTION
**Pull request by syedwaliofqual**

### CODE REVIEW

File: QualificationsRegister/Mappers/QualificationFilterNVCMapper.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            filter.MaxGuidedLearninghours = ParseInt(nvc.Get("MaxGuidedLearninghours"), "maxGuidedLearninghours");
+
+            return filter;
+        }
+
+        private static int? ParseInt(string? value, string field)
+        {
+            try
+            {
+                return value != null ? int.Parse(value!) : null;
```
Comment (obukky) [inline QualificationsRegister/Mappers/QualificationFilterNVCMapper.cs:46]: You can use Int32.tryparse here

File: QualificationsRegister/Database/RegisterDb.cs [hunk 2]
```diff
@@ -51,30 +52,121 @@ public RegisterDb(RegisterDbContext registerDbContext, ILoggerFactory loggerFact
...
+                    foreach (var qL in query.QualificationLevels)
+                    {
+                        filteredList = filteredList.Where(q => q.Level == qL);
+                    }
+                }
+
+                if (query.QualificationSubLevels != null)
+                {
+                    foreach (var qSL in query.QualificationSubLevels)
+                    {
```
Comment (obukky) [inline QualificationsRegister/Database/RegisterDb.cs:113]: https://stackoverflow.com/questions/51310113/entity-framework-core-linq-query-with-multiple-where-clauses
image

File: QualificationsRegister/Database/RegisterDb.cs [hunk 3]
```diff
@@ -132,31 +230,86 @@ public ListResponse<QualificationPublic> GetQualificationsPublicByName(int page,
...
+                }
 
-            if (!string.IsNullOrEmpty(title))
-            {
-                filteredList = quals.Where(q => q.Title.Contains(title));
+                if (query.QualificationSubLevels != null)
+                {
+                    foreach (var qSL in query.QualificationSubLevels)
+                    {
+                        filteredList = filteredList.Where(q => q.SubLevel == qSL);
```
Comment (obukky) [inline QualificationsRegister/Database/RegisterDb.cs:257]: see above

File: QualificationsRegister/Database/RegisterDb.cs (hunk 2 above)
Comment (syedwaliofqual) [reply QualificationsRegister/Database/RegisterDb.cs:113]: How would you do the foreach here?

**Review by obukky — APPROVED** [review]

**Pull request merged**